src/vos/vos_ts.c

```diff
@@ -84,13 +84,19 @@ ts_update_on_evict(struct vos_ts_table *ts_table, struct vos_ts_entry *entry)
 	}
 
 	if (other == NULL) {
-		ts_table->tt_ts_rl = MAX(ts_table->tt_ts_rl, entry->te_ts_rl);
-		ts_table->tt_ts_rh = MAX(ts_table->tt_ts_rh, entry->te_ts_rh);
+		if (entry->te_ts_rl > ts_table->tt_ts_rl) {
+			ts_table->tt_ts_rl = entry->te_ts_rl;
+			uuid_copy(ts_table->tt_tx_rl, entry->te_tx_rl);
+		}
+		if (entry->te_ts_rh > ts_table->tt_ts_rh) {
+			ts_table->tt_ts_rh = entry->te_ts_rh;
+			uuid_copy(ts_table->tt_tx_rh, entry->te_tx_rh);
+		}
 		return true;
 	}
 
-	other->te_ts_rl = MAX(other->te_ts_rl, entry->te_ts_rl);
-	other->te_ts_rh = MAX(other->te_ts_rh, entry->te_ts_rh);
+	vos_ts_rl_update(other, entry->te_ts_rl, entry->te_tx_rl);
+	vos_ts_rh_update(other, entry->te_ts_rh, entry->te_tx_rh);
```

Comment:
(style)  'te' may be misspelled - perhaps 'the'?
(style)  'te' may be misspelled - perhaps 'the'?